qcodes/parameters/parameter_base.py

```diff
@@ -307,6 +312,55 @@ def __init__(
 
             instrument.parameters[name] = self
 
+    def _build__doc__(self) -> str | None:
+        return self.__doc__
+
+    @property
+    def vals(self) -> Validator | None:
+        if len(self._vals):
+            return self._vals[0]
+        else:
+            return None
+
+    @vals.setter
+    def vals(self, vals: Validator | None) -> None:
+        if vals is not None and len(self._vals) > 0:
+            self._vals[0] = vals
+        elif vals is not None:
+            self._vals = [vals]
+        elif len(self._vals) == 1:
+            self._vals = []
+        elif len(self._vals) > 1:
+            raise RuntimeError(
+                "Cannot remove default validator from parameter with additional validators."
+            )
+        else:
+            # setting the validator to None but the parameter already doesn't have a validator
+            pass
```

Comment:
maybe it's a taste thing, but i found the below more readable as it more epxlicitly separates the None/not-None cases
```suggestion
        if vals is not None:
            if len(self._vals) > 0:
                self._vals[0] = vals
            else:
                self._vals = [vals]
        else:  # vals is None
            if len(self._vals) == 1:
                self._vals = []
            elif len(self._vals) > 1:
                raise RuntimeError(
                    "Cannot remove default validator from parameter with additional validators."
                )
            else:
                # setting the validator to None but the parameter already doesn't have a validator
                pass
```

Reply:
I don't really care either way but many other people find nested if statements a huge code smell